Point2D p0
Point2D p1
unit32 color
